api/src/test/java/org/openmrs/validator/ConceptValidatorTest.java

```diff
@@ -536,4 +558,33 @@ public void validate_shouldFailIfBlankConceptDescriptionIsPassed() throws Except
 		Assert.assertTrue(errors.hasFieldErrors("descriptions"));
 	}
 
+	/**
+	 * @see ConceptValidator#validate(Object,Errors)
+	 **/
+	@Test
+	@Verifies(value = "should fail if the concept datatype is null", method = "validate(Object,Errors)")
+	public void validate_shouldFailIfTheConceptDatatypeIsNull() throws Exception {
+		Concept concept = new Concept();
+		concept.addName(new ConceptName("some name",Context.getLocale()));
+		concept.addDescription(new ConceptDescription("some description",null));
+		concept.setConceptClass(new ConceptClass());
+		Errors errors = new BindException(concept, "concept");
+		new ConceptValidator().validate(concept, errors);
+		Assert.assertTrue(errors.hasErrors());
```

Comment:
You need to check that the datatype field has the error and that the error code is `Concept.datatype.empty`, the errors object has methods like `errors.hasFieldErrors(propertyName)` and `errors.getFieldError(propertyName).getCode()`